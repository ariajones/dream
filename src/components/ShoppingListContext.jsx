import React, { useState, createContext, useEffect } from 'react'
import firebase from 'firebase/app'

export const ShoppingListContext = createContext()

export const ShoppingListProvider = ({ children }) => {
  const [shoppingList, setShoppingList] = useState([])

  useEffect(() => {
    const unsubscribe = firebase
      .firestore()
      .collection('shoppingList')
      .onSnapshot(snapshot => {
        const newShoppingList = snapshot.docs.map(doc => ({
          id: doc.id,
          ...doc.data()
        }))
        setShoppingList(newShoppingList)
      })

    return () => unsubscribe()
  }, [])

  return (
    <ShoppingListContext.Provider value={[shoppingList, setShoppingList]}>
      {children}
    </ShoppingListContext.Provider>
  )
}
