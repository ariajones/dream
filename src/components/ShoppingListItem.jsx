import React, {useState} from 'react'
import { deleteSingleIngredient } from '../utils'

export default function ShoppingListItem ({shoppingListItem, id}) {
const [strike, setStrike]=useState({state: true, style: null})

const cross =() => {
  strike.state ? setStrike({state: false, style: {textDecoration:'line-through'}}) : setStrike({state: true, style: null })
}
  return (
    <>
      <h5 className='shoppingListIngredients' onClick={() => cross()} style={strike.style}>{shoppingListItem.quantity}{' '}{shoppingListItem.name}<span className='destroyerOfIngredients' onClick={() => deleteSingleIngredient(id)}><i class="fas fa-times"></i></span></h5>
    </>
  )
}
