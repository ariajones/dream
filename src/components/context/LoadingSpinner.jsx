import React from 'react'

export default function LoadingSpinner () {
  return (
    <>
      <div>LOADING...</div>
      <div>LOADING...</div>
      <div>LOADING...</div>
      <div>LOADING...</div>
      <div>LOADING...</div>
      <div>LOADING...</div>
      <div>LOADING...</div>
    </>
  )
}