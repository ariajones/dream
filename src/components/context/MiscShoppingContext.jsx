import React, { useState, createContext, useEffect, useContext } from 'react'

import firebase from 'firebase/app'

import { UserContext } from './UserContext'

export const MiscShoppingContext = createContext()

export const MiscShoppingProvider = ({ children }) => {
  const {user} = useContext(UserContext)
  const [miscCollection, setMiscCollection] = useState([])

  useEffect(() => {
    const unsubscribe = firebase // note unsubscribe added in case funny behaviour
      .firestore()
      .collection('miscShoppingList')
      .where('userId', '==', user ? user.uid : null)
      .onSnapshot(snapshot => {
        const miscCollection = snapshot.docs.map(doc => ({
          id: doc.id,
          ...doc.data()
        }))
        setMiscCollection(miscCollection)
      })

    return () => unsubscribe() // note unsubscribe added in case funny behaviour
  }, [user])

  return (
    <MiscShoppingContext.Provider value={[miscCollection, setMiscCollection]}>
      {children}
    </MiscShoppingContext.Provider>
  )
}
