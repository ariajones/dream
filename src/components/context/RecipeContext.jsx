import React, { useState, createContext, useEffect, useContext } from 'react'
import firebase from 'firebase/app'
import 'firebase/auth'

import { UserContext } from './UserContext'

export const RecipeContext = createContext()

export const RecipeProvider = ({ children }) => {
  const {user} = useContext(UserContext)
  const [recipes, setRecipes] = useState([])

  useEffect(() => {
    const unsubscribe = firebase 
      .firestore()
      .collection('recipes')
      .where('userId', '==', user ? user.uid : null)
      .onSnapshot(snapshot => {
        const newRecipes = snapshot.docs.map(doc => ({
          id: doc.id,
          ...doc.data()
        }))
        setRecipes(newRecipes)
      })

    return () => unsubscribe()
  }, [user])

  return (
    <RecipeContext.Provider value={[recipes, setRecipes]}>
      {children}
    </RecipeContext.Provider>
  )
}