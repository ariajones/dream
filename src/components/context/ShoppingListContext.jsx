import React, { useState, createContext, useEffect } from 'react'

import firebase from 'firebase/app'

export const ShoppingListContext = createContext()

export const ShoppingListProvider = ({ children }) => {
  const [shoppingList, setShoppingList] = useState([])

  useEffect(() => {
    const unsubscribe = firebase // note unsubscribe added in case funny behaviour
      .firestore()
      .collection('shoppingList')
      .onSnapshot(snapshot => {
        const newShoppingList = snapshot.docs.map(doc => ({
          id: doc.id,
          ...doc.data()
        }))
        setShoppingList(newShoppingList)
      })

    return () => unsubscribe() // note unsubscribe added in case funny behaviour
  }, [])

  return (
    <ShoppingListContext.Provider value={[shoppingList, setShoppingList]}>
      {children}
    </ShoppingListContext.Provider>
  )
}
