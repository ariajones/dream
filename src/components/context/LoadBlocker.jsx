import React, { useContext} from 'react'

import { RecipeContext } from './RecipeContext'
import { SelectedDayContext } from './SelectedDayContext'
import { ShoppingListContext } from './ShoppingListContext'
import { StrikeOut } from './StrikeoutContext'
import { UserContext } from './UserContext'
import { WeekContext } from './WeekContext'
import LoadingSpinner from './LoadingSpinner'


export default function LoaderBlocker ({ children }) {
  const {user} = useContext(UserContext)
  const [recipes] = useContext(RecipeContext)
  const [selectedDay] = useContext(SelectedDayContext)
  const [shoppingList] = useContext(ShoppingListContext)
  const [week] = useContext(WeekContext)
  // const [strikeOut] = useContext(StrikeOut)

  if (!user || !recipes || !selectedDay || !shoppingList || !week) {
    return <LoadingSpinner />
  } 
    return children
}