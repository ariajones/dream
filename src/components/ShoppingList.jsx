import React, { useContext, useState } from 'react'

import { UserContext } from './context/UserContext'
import { ShoppingListContext } from './context/ShoppingListContext'
import ShoppingListItem from './ShoppingListItem'
import { clearShoppingList, clearMiscShoppingList, addMiscItem, sortList } from '../utils'


function ShoppingList () {
  const {user} = useContext(UserContext)
  const [shoppingList] = useContext(ShoppingListContext)
  const [sortBy, setSortBy] = useState('NAME_ASC')
  const [newItem, setNewItem] = useState('')
  const [newItemQ, setNewItemQ] = useState('')
  const sortedShoppingList = sortList([...shoppingList], sortBy)

  const clearClickHandler = () => {
    clearShoppingList(user.uid)
    clearMiscShoppingList(user.uid)
  }
  
  const submitHandler = (e) => {
    e.preventDefault()
    if (newItem) {
    addMiscItem(user.uid, newItemQ, newItem)
    setNewItem('')
    setNewItemQ('')
    }
  }
  
  console.log('shopping list', shoppingList)


  return (
    <div className="ShoppingList">
      <button onClick={() => clearClickHandler()}>Clear Shopping List</button>
      <label>Sort By:</label>{' '}
        <select className='recipeSortButton'  value={sortBy} onChange={evt => setSortBy(evt.target.value)}>
          <option value='NAME_ASC'>Name (A-Z)</option>
          <option value='NAME_DESC'>Name (Z-A)</option>
        </select>
      <form>
        {/* <label>Add item to Shopping List</label><br></br> */}
        <input className='noBulmaBorderless newItemQ' type="number" placeholder='Quantity' value={newItemQ} onChange={e => setNewItemQ(e.currentTarget.value)} />{' '}
        <input className='noBulmaBorderless newItem' type="text" placeholder='Item, ie bananas' value={newItem} onChange={e => setNewItem(e.currentTarget.value)} />{' '}
        <h3 onClick={e => submitHandler(e)} style={{display: 'inline'}}><h3 className='plusButton' >+</h3>Add item</h3>
        {/* <button className='noBulmaButton'>Add</button> */}
      </form>
      <ul>
        {sortedShoppingList.map(shoppingListItem =>
          <li key={shoppingListItem.id}>
            <ShoppingListItem id={shoppingListItem.id} shoppingListItem={shoppingListItem}/>
          </li>
        )}
      </ul>
      <button onClick={() => clearClickHandler()}>Clear Shopping List</button>
    </div>
  )
}

export default ShoppingList
