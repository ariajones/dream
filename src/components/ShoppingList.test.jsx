import React, { useState, useContext } from 'react'
import renderer from 'react-test-renderer'

import ShoppingList from './ShoppingList'

import { ShoppingListContext } from './context/ShoppingListContext'

import ShoppingListItem from './ShoppingListItem'

export const FakeShoppingListProvider = ({ children }) => {
    const [shoppingList, setShoppingList] = useState([{
        id: "ShoppingListCollectionId",
        ingredients: ["1½ red onions", "100 g rye bread", "ground corian…"],
        recipeId: "testRecipeId"
    }])

    return (
        <ShoppingListContext.Provider value={[shoppingList, setShoppingList]}>
          {children}
        </ShoppingListContext.Provider>
      )
}

it ('renders component correctly', () => {
    const dream = renderer
    .create(
        <FakeShoppingListProvider>
                <ShoppingList>
                    <ShoppingListItem/>
                </ShoppingList>
        </FakeShoppingListProvider>
    )
    .toJSON()
expect(dream).toMatchSnapshot()
})