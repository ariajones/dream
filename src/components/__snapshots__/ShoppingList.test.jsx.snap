// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component correctly 1`] = `
<div
  className="ShoppingList"
>
  <div>
     
  </div>
  <div
    className="sort-sort"
  >
    <button
      className="button is-small is-rounded is-outline clear-shopping"
      onClick={[Function]}
    >
      Clear all
    </button>
    <select
      className="recipeSortButton"
      onChange={[Function]}
      value="NAME_ASC"
    >
      <option
        value="NAME_ASC"
      >
        Name (A-Z)
      </option>
      <option
        value="NAME_DESC"
      >
        Name (Z-A)
      </option>
    </select>
  </div>
  <div
    className="search-pad"
  >
    <form
      className="shopping-list"
    >
      <input
        className="noBulmaBorderless newItemQ"
        onChange={[Function]}
        placeholder="Quantity"
        type="number"
        value=""
      />
       
      <input
        className="noBulmaBorderless newItem"
        onChange={[Function]}
        placeholder="Item, ie bananas"
        type="text"
        value=""
      />
       
      <h3
        onClick={[Function]}
        style={
          Object {
            "display": "inline",
          }
        }
      >
        <h3
          className="plusButton"
        >
          +
        </h3>
        Add
      </h3>
    </form>
  </div>
  <div
    className="shopping-list"
  >
    <ul>
      <li>
        <h5
          className="shoppingListIngredients"
          onClick={[Function]}
          style={null}
        >
           
          <span
            className="destroyerOfIngredients"
            onClick={[Function]}
          >
            <i
              class="fas fa-times"
            />
          </span>
        </h5>
      </li>
    </ul>
  </div>
</div>
`;
